src/Rule/UnusedVariable.php

```diff
@@ -25,7 +25,11 @@ public function check(TokenStream $tokens)
                 $blockType = $tokens->look(2)->getValue();
 
                 if (in_array($blockType, ['block', 'for', 'embed', 'macro'], true)) {
```

Comment:
shouldn't `with` also be in this list ? It is specifically about creating a new scope.